test/module/ecdsa-registry-module.specs.ts

```diff
@@ -0,0 +1,156 @@
+
+
+            let EcdsaOwnershipSetupData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [await randomContract.address]
+              );
+
+            // Getting the address of the to be deployed SA does not get revert as it's just a bunch of abi.encode operations
+            const expectedSmartAccountAddress = await SAFactory.getAddressForCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            // This should revert
+            await expect(SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex)).to.reverted;
+            // Extra checks
+            // smartAccountOwners mapping should point to address(0)
+            expect( await EcdsaRegistryModule.smartAccountOwners(expectedSmartAccountAddress)).to.be.equal(AddressZero);
+        });
+
+        // Listen for AlreadyInitedForSmartAccount(msg.sender) error
+        it("Reverts when calling again after initialization", async()=>{
+            const {SAFactory,EcdsaRegistryModule,entryPoint} = await setupTests();
+
+            let EcdsaOwnershipSetupData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [await smartAccountOwner.getAddress()]
+              );
+
+            const expectedSmartAccountAddress = await SAFactory.getAddressForCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            await SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            const userSA = await hre.ethers.getContractAt("SmartAccount", expectedSmartAccountAddress);
+
+            await deployer.sendTransaction({
+                to: userSA.address,
+                value: ethers.utils.parseEther("10"),
+            });
+
+            // Send txs to ECDSARegistryModule.initForSmartAccount and check for revert
+            // Construct userOp
+            const txnData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [bob.getAddress()],
+            );
+            const userOp = await fillAndSign(
+                {
+                    sender: userSA.address,
+                    callData: txnData
+                },
+                smartAccountOwner,
+                entryPoint,
+                'nonce'
+            );
+            let signatureWithModuleAddress = ethers.utils.defaultAbiCoder.encode(
+                ["bytes", "address"],
+                [userOp.signature, EcdsaRegistryModule.address]
+            );
+            userOp.signature = signatureWithModuleAddress;
+
+            const tx = await entryPoint.handleOps([userOp], alice.address);
+            await expect(tx).to.not.emit(entryPoint, "UserOperationRevertReason");
```

Comment:
if 'UserOperationRevertReason' is **not** emitted, that means there's no revert.

Reply:
this was a silly mistake by my side :(